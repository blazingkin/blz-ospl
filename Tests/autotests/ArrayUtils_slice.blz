:main
    #Test 00 :slice
	print(">>> Test 00 slice <<<")
	arr = [0,1,2,3,4,5,{nil},7,8]
	print("arr: " + arr)
	arr1 = arr.slice(0,3)
	print("arr1 sliced from arr: " + arr1)
	arr2 = arr.slice(0,arr.length())
	print("arr2 sliced from arr: " + arr2)
	arr3 = arr.slice(0,0)
	print("arr3 sliced from arr: " + arr3)
	arr1 = arr.slice(0,1)
	print("arr1 sliced from arr1: " + arr1)
	print("--------------------------------")

	#does not really make sense to use slice!()
	
end