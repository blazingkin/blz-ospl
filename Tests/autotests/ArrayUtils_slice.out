>>> Test 00 slice <<<
arr: [0, 1, 2, 3, 4, 5, , 7, 8]
arr1 sliced from arr: [0, 1, 2]
arr2 sliced from arr: [0, 1, 2, 3, 4, 5, , 7, 8]
arr3 sliced from arr: []
arr1 sliced from arr1: [0]
--------------------------------
